src/error.ts

```diff
@@ -692,9 +713,9 @@ export class MongoMissingCredentialsError extends MongoAPIError {
  * @category Error
  */
 export class MongoMissingDependencyError extends MongoAPIError {
-  constructor(message: string, { cause }: { cause?: Error } = {}) {
-    super(message);
-    if (cause) this.cause = cause;
+  /** @internal */
+  constructor(message: string, options: { cause?: Error } = {}) {
+    super(message, options);
   }
```

Comment:
What exactly is it that adding `@internal` is supposed to achieve? It won’t prevent users from creating class instances, it’s only going to make the class look like it doesn’t have its own constructor, i.e. uses the `constructor(...args) { super(...args); }` one.

If you want to prevent users from creating instances of these classes, you may need to do something like

```js
export class MongoSomethingError extends MongoError  {
  private constructor(...) {}

  /** @internal */
  static create(...args) { return new this(...args); }
}
```

Reply:
We're doing this in an attempt to communicate that these classes shouldn't be instantiated by users. It's a pattern we have a lot elsewhere in the driver that probably could use some work with regards to getting that message across.

Wouldn't this solution also have the same effect? The user might not have documentation for the static `create` method, but they would still be able to call it on their end, right?

Reply:
Yeah, I get the intention here, and I agree that it's a goal that's worth putting effort into. :)

The problem is specifically with marking constructors as `@internal`, which is _different_ from all other methods, because that it does _not_ make it appear to consumers of your .d.ts file that there is no publicly accessible constructor; instead, the TS definition effectively says that "the constructor of this class has the same signature as the  constructor of the parent class".

That's different with a static method; yeah, of course users could still call it by casting the class to `as any` or just not using TS validation at all, but at least it would not show up in the generated TS definitions.

Reply:
Okay, I see what you're saying now. I'll check with the team to see if we want to start with that new pattern here or deal with it in a separate ticket that takes care of this across the driver. Thanks for the clarification!